extensions/firestore-huggingface-inference-api/functions/src/inference.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  ConversationalOutput,
+  FeatureExtractionOutput,
+  FillMaskOutput,
+  HfInference,
+  HfInferenceEndpoint,
+  QuestionAnsweringOutput,
+  SentenceSimilarityOutput,
+  SummarizationOutput,
+  TableQuestionAnsweringOutput,
+  TextClassificationOutput,
+  TextGenerationOutput,
+  TokenClassificationOutput,
+  TranslationOutput,
+  ZeroShotClassificationOutput,
+} from '@huggingface/inference';
+import * as functions from 'firebase-functions';
+
+import * as tasks from './tasks';
+import { TaskId } from './tasks';
+
+type InferenceOutput =
+  | FillMaskOutput
+  | SummarizationOutput
+  | QuestionAnsweringOutput
+  | TableQuestionAnsweringOutput
+  | SentenceSimilarityOutput
+  | TextClassificationOutput
+  | TextGenerationOutput
+  | TokenClassificationOutput
+  | TranslationOutput
+  | ZeroShotClassificationOutput
+  | FeatureExtractionOutput
+  | ConversationalOutput;
+
+/**
+ * Validate the input and run inference.
+ *
+ * @param {functions.firestore.DocumentSnapshot} snapshot The Firestore snapshot.
+ * @return {InferenceOutput} The inference output based on the specified task.
+ */
+export async function runInference(
+  snapshot: functions.firestore.DocumentSnapshot,
+  task: TaskId,
+  inference: HfInference | HfInferenceEndpoint,
+): Promise<InferenceOutput> {
+  switch (task) {
+    case TaskId.fillMask: {
+      return await tasks.fillMask(snapshot, inference);
```

Comment:
I don't believe you need `return` and await `here`. return should automatically recognise the promise